agent/build.gradle.kts

```diff
@@ -29,9 +29,10 @@ val upstreamAgent: Configuration by configurations.creating {
   isCanBeConsumed = false
 }
 
-val otelInstrumentationVersion: String by extra
+val otelInstrumentationVersion: String by rootProject.extra
 
 dependencies {
+  add("upstreamAgent", platform(project(":dependencyManagement")))
```

Comment:
Does the loop in the main `build.gradle.kts` file not handle this case?

Reply:
Yeah this surprised me as well, but in practice, no, it did not handle it. I'm I assume it is caused by the fact that those configurations are created/defined local to the agent subproject build script? There are a couple other places that required this as well.

Reply:
Gradle black magic I guess 🧙 💀